packages/notifi-react-card/lib/hooks/useNotificationHistory.tsx

```diff
@@ -0,0 +1,149 @@
+import {
+  AccountBalanceChangedEventDetails,
+  BroadcastMessageEventDetails,
+  ChatMessageReceivedEventDetails,
+  GenericEventDetails,
+  HealthValueOverThresholdEventDetails,
+  NotificationHistoryEntry,
+} from '@notifi-network/notifi-core';
+import React from 'react';
+
+import { AnnouncementIcon } from '../assets/AnnouncementIcon';
+import { ChatAlertIcon } from '../assets/ChatAlertIcon';
+import { RatioCheckIcon } from '../assets/RatioCheckIcon';
+import { SwapIcon } from '../assets/SwapIcon';
+import { AlertIcon } from '../components/AlertHistory/AlertIcon';
+import { AlertNotificationViewProps } from '../components/AlertHistory/AlertNotificationRow';
+import { formatAmount } from '../utils/AlertHistoryUtils';
+
+export const useNotificationHistory = () => {
+  class SupportedEntryDetail {
+    typename: string;
+    generateProps: (
+      notification: NotificationHistoryEntry,
+    ) => AlertNotificationViewProps;
+    constructor(
+      typename: string,
+      generateProps: (
+        notification: NotificationHistoryEntry,
+      ) => AlertNotificationViewProps,
+    ) {
+      this.typename = typename;
+      this.generateProps = generateProps;
+    }
+  }
+
+  const supportedHistoryDetails: SupportedEntryDetail[] = [
```

Comment:
it would be better to see this as a Record<string, SupportedEntryDetail | undefined> so that `.find` is O[1]